dropwizard-index-page/src/main/java/com/palantir/indexpage/FileSystemIndexPage.java

```diff
@@ -46,42 +48,31 @@ public FileSystemIndexPage(File file) {
     }
 
     @Override
-    public Optional<String> getContent(Map<String, String> inputContext) {
-        checkNotNull(inputContext);
-
+    public Optional<String> getContent() {
         try {
-            checkAndReloadFile(inputContext);
+            checkAndReloadFile();
         } catch (FileNotFoundException e) {
 
             // the file could be deleted during run time, so return absent in order to handle the behavior gracefully
             return Optional.absent();
         }
 
-        return Optional.of(writer.toString());
+        return Optional.of(content);
     }
 
-    private synchronized void checkAndReloadFile(Map<String, String> inputContext) throws FileNotFoundException {
+    private synchronized void checkAndReloadFile() throws FileNotFoundException {
 
-        // If file has been changed, we need to load the file and resolve the template. Otherwise,
-        // only resolve the template if the context has been changed.
+        // If file has been changed, we need to reload the file and resolve the template.
         if (lastModified != file.lastModified()) {
             loadFile();
-            executeTemplate(inputContext);
-        } else if (!inputContext.equals(context)) {
-            executeTemplate(inputContext);
         }
     }
 
-    private void loadFile() throws FileNotFoundException {
-        BufferedReader reader;
-        reader = Files.newReader(file, Charsets.UTF_8);
-        mustacheCompiler = factory.compile(reader, IndexPageBundle.INDEX_PAGE_NAME);
+    private synchronized void loadFile() throws FileNotFoundException {
+        mustacheCompiler = factory.compile(Files.newReader(file, Charsets.UTF_8), IndexPageBundle.INDEX_PAGE_NAME);
```

Comment:
does this need to be saved each time? it seems like it's a new one every `loadFile` call